ui/address/details/AddressBalance.tsx

```diff
@@ -65,6 +75,66 @@ const AddressBalance = ({ data, isLoading }: Props) => {
     handler: handleNewCoinBalanceMessage,
   });
 
+  const value = data.coin_balance || '0';
+  const exchangeRate = data.exchange_rate;
+  const decimals = String(config.chain.currency.decimals);
+  const accuracyUsd = 2;
+  const accuracy = 8;
+
+  const onGetGasClick = React.useCallback(() => {
+    mixpanel.logEvent(mixpanel.EventTypes.BUTTON_CLICK, { Content: 'Get gas', Source: 'address' });
+  }, []);
+
+  let getGasButton = null;
+  const { usd: usdResult } = getCurrencyValue({ value, accuracy, accuracyUsd, exchangeRate, decimals });
+
+  if (getGasFeature.isEnabled && !data?.is_contract && Number(usdResult) < getGasFeature.usdThreshold) {
+    const buttonContent = (
+      <>
+        { getGasFeature.logoUrl && (
+          <Image src={ getGasFeature.logoUrl } alt={ getGasFeature.name } boxSize={ 5 } mr={ 2 } borderRadius="4px" overflow="hidden"/>
+        ) }
+        { getGasFeature.name }
+      </>
+    );
+
+    const linkProps = {
+      display: 'flex',
+      alignItems: 'center',
+      fontSize: 'sm',
+      lineHeight: 5,
+      onClick: onGetGasClick,
+    };
```

Comment:
Instead of this, maybe it is better a conditional Link component:

```js
const Link = typeof dappId === 'string' ? LinkInternal : LinkExternal;
```